applications/luci-ocserv/luasrc/model/cbi/ocserv/main.lua

```diff
@@ -27,6 +27,28 @@ local e = s:taboption("general", Flag, "enable", translate("Enable server"))
 e.rmempty = false
 e.default = "1"
 
+local o = s:taboption("general", Value, "sha_hash", translate("Server's certificate SHA1 hash"))
+local fd = io.popen("/usr/bin/certtool -i --infile /etc/ocserv/server-cert.pem |/bin/grep -A 1 SHA1|/usr/bin/tail -1|/bin/sed 's/\t//g'|/usr/bin/tr 'a-z' 'A-Z'", "r")
+if fd then local ln
+	ln = fd:read("*l")
+	fd:close()
+
+	if ln then
+		o.placeholder = ln
+	end
+end
+
+local o = s:taboption("general", Value, "pkid", translate("Server's Public Key ID"))
+local fd = io.popen("/usr/bin/certtool -i --infile /etc/ocserv/server-cert.pem |/bin/grep -A 1 'Public Key ID'|/usr/bin/tail -1|/bin/sed 's/\t//g'|/usr/bin/tr 'a-z' 'A-Z'", "r")
+if fd then local ln
```

Comment:
Same here:
    if fd then
            local found = false
            while true do
                    local ln = fd:read("*l")
                    if not ln then
                            break
                    elseif ln:match("Public Key I[Dd]:") then
                            found = true
                    elseif found then
                            local hash = ln:match("([a-f0-9]+)")
                            o.placeholder = hash and hash:upper()
                            break
                    end
            end

            fd:close()
    end